### ISSUE TITLE
Building grpc with bazel by following default instructions fails finding basic c++ headers for clang (with gcc-12 installed)

### ISSUE THREAD

**Issue by tapetersen** [grpc/grpc#38018]
### TL;DR; For someone just running into the problem.
Narrowed down this case to the problem of having  gcc-12 installed without libstdc++-12-dev. 
Try installing the libstd++ headers instead and do a clean before rebuilding (or uninstall gcc-12)
```
$ sudo apt install libstdc++-12-dev
$ bazel clean --expunge
```

### What version of gRPC and what language are you using?
I've cloned the repository from here and checked out the  latest tag
```
commit d3286610f703a339149c3f9be69f0d7d0abb130a (HEAD, tag: v1.67.1, upstream/v1.67.x)
Author: Sreenithi Sridharan <ssreenithi@google.com>
Date:   Mon Oct 28 21:57:38 2024 +0530

    [Release] Bump version to 1.67.1 (on v1.67.x branch) (#38011)
    
    Change was created by the release automation script. See go/grpc-release
```

### What operating system (Linux, Windows,...) and version?
Ubuntu 22.04.5 LTS

### What runtime / compiler are you using (e.g. python version or version of gcc)
```
$ clang --version
Ubuntu clang version 14.0.0-1ubuntu1.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin
```

```
$ bazelisk version
Bazelisk version: v1.22.1
INFO: Running bazel wrapper (see //tools/bazel for details), bazel version 7.3.1 will be used instead of system-wide bazel installation.
Build label: 7.3.1
Build target: @@//src/main/java/com/google/devtools/build/lib/bazel:BazelServer
Build time: Mon Aug 19 16:12:50 2024 (1724083970)
Build timestamp: 1724083970
Build timestamp as int: 1724083970
```
### What did you do?

TL;DR;
I tried building the repository with bazel following the instructions listed but ran into problems probably caused by/related to https://github.com/bazelbuild/bazel/issues/13071

It looks like it was solved with https://github.com/bazelbuild/bazel/pull/13666 but I have no experience with bazel and version so I guess it could be because grpc is using an older bazel version where it's not included.

#### Longer reproducing steps and info

I tried to follow the steps for cloning and building the repository as documented starting here https://github.com/grpc/grpc/blob/master/CONTRIBUTING.md

I mostly followed the clone instructions here https://github.com/grpc/grpc/blob/master/BUILDING.md#clone-the-repository-including-submodules

With the simple difference of cloning the full repository and then doing a checkout of the tag
```
$ cd grpc
$ git checkout v1.67.1
$ git submodule update --init
```

I tried to follow the instructions under "Building with bazel" https://github.com/grpc/grpc/blob/master/BUILDING.md#building-with-bazel-recommended

I downloaded the deb package for bazilisk 1.22.1 `bazelisk-amd64.deb` (latest as of now) from the the releases page https://github.com/bazelbuild/bazelisk/releases/tag/v1.22.1 and installed with apt

```
sudo apt install Dowloads/bazelisk-amd64.deb
```

I installed the prerequisites listed here https://github.com/grpc/grpc/blob/master/BUILDING.md#clone-the-repository-including-submodules

```
sudo apt install build-essential autoconf libtool pkg-config cmake clang libc++-dev
```

I then tried to build everything with 
```
$ bazel build :all
```

### What did you expect to see?
A successfully built project

### What did you see instead?

Errors about clang failing to find basic stdlib headers. (different everytime but always something from stdlib)

```
$ bazel build :all
INFO: Running bazel wrapper (see //tools/bazel for details), bazel version 7.3.1 will be used instead of system-wide bazel installation.
Starting local Bazel server and connecting to it...
WARNING: /home/tobias/.cache/bazel/_bazel_tobias/8212f703b0c24e14895c9c706b73a907/external/google_cloud_cpp/google/cloud/BUILD.bazel:53:15: in config_setting rule @@google_cloud_cpp//google/cloud:enable_opentelemetry_valid-transition: target '@@google_cloud_cpp//google/cloud:enable_opentelemetry_valid-transition' depends on deprecated target '@@google_cloud_cpp//:enable-experimental-opentelemetry': this flag is now GA, please use //:enable_opentelemetry instead.
WARNING: /home/tobias/.cache/bazel/_bazel_tobias/8212f703b0c24e14895c9c706b73a907/external/google_cloud_cpp/google/cloud/BUILD.bazel:62:15: in config_setting rule @@google_cloud_cpp//google/cloud:disable_opentelemetry: target '@@google_cloud_cpp//google/cloud:disable_opentelemetry' depends on deprecated target '@@google_cloud_cpp//:enable-experimental-opentelemetry': this flag is now GA, please use //:enable_opentelemetry instead.
ERROR: /home/tobias/.cache/bazel/_bazel_tobias/8212f703b0c24e14895c9c706b73a907/external/com_google_absl/absl/strings/BUILD.bazel:49:11: Compiling absl/strings/internal/charconv_bigint.cc failed: (Exit 1): clang failed: error executing CppCompile command (from target @@com_google_absl//absl/strings:strings) /usr/lib/llvm-14/bin/clang -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer ... (remaining 78 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/com_google_absl/absl/strings/internal/charconv_bigint.cc:15:
external/com_google_absl/absl/strings/internal/charconv_bigint.h:18:10: fatal error: 'algorithm' file not found
#include <algorithm>
         ^~~~~~~~~~~
1 error generated.
ERROR: /home/tobias/.cache/bazel/_bazel_tobias/8212f703b0c24e14895c9c706b73a907/external/com_google_absl/absl/strings/BUILD.bazel:49:11: Compiling absl/strings/internal/charconv_parse.cc failed: (Exit 1): clang failed: error executing CppCompile command (from target @@com_google_absl//absl/strings:strings) /usr/lib/llvm-14/bin/clang -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer ... (remaining 78 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/com_google_absl/absl/strings/internal/charconv_parse.cc:15:
external/com_google_absl/absl/strings/internal/charconv_parse.h:18:10: fatal error: 'cstdint' file not found
#include <cstdint>
         ^~~~~~~~~
1 error generated.
ERROR: /home/tobias/.cache/bazel/_bazel_tobias/8212f703b0c24e14895c9c706b73a907/external/com_google_absl/absl/strings/BUILD.bazel:49:11: Compiling absl/strings/internal/damerau_levenshtein_distance.cc failed: (Exit 1): clang failed: error executing CppCompile command (from target @@com_google_absl//absl/strings:strings) /usr/lib/llvm-14/bin/clang -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer ... (remaining 78 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/com_google_absl/absl/strings/internal/damerau_levenshtein_distance.cc:15:
external/com_google_absl/absl/strings/internal/damerau_levenshtein_distance.h:18:10: fatal error: 'cstdint' file not found
#include <cstdint>
         ^~~~~~~~~
1 error generated.
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 10.359s, Critical Path: 0.53s
INFO: 41 processes: 41 internal.
ERROR: Build did NOT complete successfully
```

Adding the verbose-failure flag:
```
 bazel build :all  --verbose_failures
INFO: Running bazel wrapper (see //tools/bazel for details), bazel version 7.3.1 will be used instead of system-wide bazel installation.
WARNING: /home/tobias/.cache/bazel/_bazel_tobias/8212f703b0c24e14895c9c706b73a907/external/google_cloud_cpp/google/cloud/BUILD.bazel:62:15: in config_setting rule @@google_cloud_cpp//google/cloud:disable_opentelemetry: target '@@google_cloud_cpp//google/cloud:disable_opentelemetry' depends on deprecated target '@@google_cloud_cpp//:enable-experimental-opentelemetry': this flag is now GA, please use //:enable_opentelemetry instead.
WARNING: /home/tobias/.cache/bazel/_bazel_tobias/8212f703b0c24e14895c9c706b73a907/external/google_cloud_cpp/google/cloud/BUILD.bazel:53:15: in config_setting rule @@google_cloud_cpp//google/cloud:enable_opentelemetry_valid-transition: target '@@google_cloud_cpp//google/cloud:enable_opentelemetry_valid-transition' depends on deprecated target '@@google_cloud_cpp//:enable-experimental-opentelemetry': this flag is now GA, please use //:enable_opentelemetry instead.
ERROR: /home/tobias/.cache/bazel/_bazel_tobias/8212f703b0c24e14895c9c706b73a907/external/com_google_absl/absl/strings/BUILD.bazel:118:11: Compiling absl/strings/internal/utf8.cc failed: (Exit 1): clang failed: error executing CppCompile command (from target @@com_google_absl//absl/strings:internal) 
  (cd /home/tobias/.cache/bazel/_bazel_tobias/8212f703b0c24e14895c9c706b73a907/sandbox/linux-sandbox/19/execroot/com_github_grpc_grpc && \
  exec env - \
    GRPC_BAZEL_RUNTIME=1 \
    PATH=/home/tobias/Projects/grpc/tools:/home/tobias/.local/bin:/home/tobias/.local/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin \
    PWD=/proc/self/cwd \
  /usr/lib/llvm-14/bin/clang -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer '-std=c++14' -MD -MF bazel-out/k8-fastbuild/bin/external/com_google_absl/absl/strings/_objs/internal/utf8.pic.d '-frandom-seed=bazel-out/k8-fastbuild/bin/external/com_google_absl/absl/strings/_objs/internal/utf8.pic.o' -fPIC -iquote external/com_google_absl -iquote bazel-out/k8-fastbuild/bin/external/com_google_absl -Wno-typedef-redefinition -DGRPC_BAZEL_BUILD '-std=c++14' -Wall -Wextra -Wcast-qual -Wconversion -Wfloat-overflow-conversion -Wfloat-zero-conversion -Wfor-loop-analysis -Wformat-security -Wgnu-redeclared-enum -Winfinite-recursion -Winvalid-constexpr -Wliteral-conversion -Wmissing-declarations -Woverlength-strings -Wpointer-arith -Wself-assign -Wshadow-all -Wshorten-64-to-32 -Wsign-conversion -Wstring-conversion -Wtautological-overlap-compare -Wtautological-unsigned-zero-compare -Wundef -Wuninitialized -Wunreachable-code -Wunused-comparison -Wunused-local-typedefs -Wunused-result -Wvla -Wwrite-strings -Wno-float-conversion -Wno-implicit-float-conversion -Wno-implicit-int-float-conversion -Wno-unknown-warning-option -DNOMINMAX -no-canonical-prefixes -Wno-builtin-macro-redefined '-D__DATE__="redacted"' '-D__TIMESTAMP__="redacted"' '-D__TIME__="redacted"' '-fmodule-name=com_google_absl//absl/strings:internal' '-fmodule-map-file=bazel-out/k8-fastbuild/bin/external/com_google_absl/absl/strings/internal.cppmap' -Xclang -fno-cxx-modules -fmodules-strict-decluse -Wprivate-header '-fmodule-map-file=external/local_config_cc/module.modulemap' '-fmodule-map-file=bazel-out/k8-fastbuild/bin/external/com_google_absl/absl/base/config.cppmap' '-fmodule-map-file=bazel-out/k8-fastbuild/bin/external/com_google_absl/absl/base/core_headers.cppmap' '-fmodule-map-file=bazel-out/k8-fastbuild/bin/external/com_google_absl/absl/base/endian.cppmap' '-fmodule-map-file=bazel-out/k8-fastbuild/bin/external/com_google_absl/absl/base/raw_logging_internal.cppmap' '-fmodule-map-file=bazel-out/k8-fastbuild/bin/external/com_google_absl/absl/meta/type_traits.cppmap' -c external/com_google_absl/absl/strings/internal/utf8.cc -o bazel-out/k8-fastbuild/bin/external/com_google_absl/absl/strings/_objs/internal/utf8.pic.o)
# Configuration: 8915d6d8236493a7847c797848c67550f8e3b519540dcc91cc75881196a5cbed
# Execution platform: @@internal_platforms_do_not_use//host:host

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
In file included from external/com_google_absl/absl/strings/internal/utf8.cc:17:
external/com_google_absl/absl/strings/internal/utf8.h:20:10: fatal error: 'cstddef' file not found
#include <cstddef>
         ^~~~~~~~~
1 error generated.
INFO: Elapsed time: 2.683s, Critical Path: 0.33s
INFO: 10 processes: 10 internal.
ERROR: Build did NOT complete successfully
```

See [TROUBLESHOOTING.md](https://github.com/grpc/grpc/blob/master/TROUBLESHOOTING.md) for how to diagnose problems better.

**Comment by tapetersen** [comment]
it does sound like there is a somewhat straightforward workaround as described in https://github.com/bazelbuild/bazel/pull/13666 but with my limited bazel knowledge I can't decipher anything that would simply get the build going.

**Comment by tapetersen** [comment]
Ok after stumbling around a bit more and reading things a bit more I tried installing the `libstdc++` headers instead (`sudo apt install libstdc++-12-dev` and cleaning with `bazel clean --expunge`  it works here at least.


If nothing else I would suggest an update to the notes in https://github.com/grpc/grpc/blob/master/BUILDING.md#linux at least (not sure what version as there doesn't seem to exist a version-less meta-package for in the same way as for `libc++`

**Comment by tapetersen** [comment]
You can consider the issue  for my part resolved but some note in some documentation would probably be good as I don't think my build environment is that uncommon. I'm just not sure how specific to the Ubuntu setup this is and so where it belongs.

**Comment by yijiem** [comment]
I'm not able to reproduce this. I made a Dockerfile here:

```
FROM ubuntu:20.04

ENV DEBIAN_FRONTEND=noninteractive

RUN apt-get -y update && apt-get install -y git wget curl
RUN apt-get install -y build-essential autoconf libtool pkg-config cmake clang libc++-dev
RUN wget https://github.com/bazelbuild/bazelisk/releases/download/v1.22.1/bazelisk-linux-amd64
RUN chmod 755 /bazelisk-linux-amd64
RUN mv /bazelisk-linux-amd64 /usr/bin/bazelisk

RUN groupadd -g 1234 yijiem && \
    useradd -m -u 1234 -g yijiem yijiem

USER yijiem
WORKDIR /home/yijiem

RUN git clone https://github.com/grpc/grpc.git

WORKDIR /home/yijiem/grpc
RUN git checkout v1.67.1
RUN git submodule update --init

CMD ["/bin/bash"]
```

Build it: `$ sudo docker build --no-cache -t 38018 -f Dockerfile .`
Run it: `$ sudo docker run -it --rm 38018`
And then run inside the container: `$ bazelisk build :all`

I don't see the said error.

**Comment by tapetersen** [comment]
Ok thanks for a great response and I'll see if I can dump my packages and narrow down the issue. Is there any good flags to bazel to debug it's compiler toolchain resolve (especially the include dirs  for clang)

**Comment by tapetersen** [comment]
Also I note you used ubuntu 20.04 while I'm on 22.04 if that may make the difference

**Comment by tapetersen** [comment]
Ok digging further with some test files it's probably related to my local computer having gcc 12 installed as well as 11

locally
```
$ /usr/lib/llvm-14/bin/clang++ -H test.cc  -v
Ubuntu clang version 14.0.0-1ubuntu1.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-14/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/11
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/12
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/12
Candidate multilib: .;@m64
Selected multilib: .;@m64
 "/usr/lib/llvm-14/bin/clang" -cc1 -triple x86_64-pc-linux-gnu -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name test.cc -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -debugger-tuning=gdb -v -H -sys-header-deps -fcoverage-compilation-dir=/home/tobias/Projects/grpc -resource-dir /usr/lib/llvm-14/lib/clang/14.0.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++ -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/x86_64-linux-gnu -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/backward -internal-isystem /usr/lib/llvm-14/lib/clang/14.0.0/include -internal-isystem /usr/local/include -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/12/../../../../x86_64-linux-gnu/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -fdeprecated-macro -fdebug-compilation-dir=/home/tobias/Projects/grpc -ferror-limit 19 -fgnuc-version=4.2.1 -fcxx-exceptions -fexceptions -fcolor-diagnostics -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /tmp/test-f4b181.o -x c++ test.cc
clang -cc1 version 14.0.0 based upon LLVM 14.0.0 default target x86_64-pc-linux-gnu
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/x86_64-linux-gnu"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++/backward"
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/12/../../../../x86_64-linux-gnu/include"
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/12/../../../../include/c++
 /usr/lib/llvm-14/lib/clang/14.0.0/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
 "/usr/bin/ld" -pie -z relro --hash-style=gnu --build-id --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o a.out /lib/x86_64-linux-gnu/Scrt1.o /lib/x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/12/crtbeginS.o -L/usr/lib/gcc/x86_64-linux-gnu/12 -L/usr/lib/gcc/x86_64-linux-gnu/12/../../../../lib64 -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib64 -L/usr/lib/llvm-14/bin/../lib -L/lib -L/usr/lib /tmp/test-f4b181.o -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/lib/gcc/x86_64-linux-gnu/12/crtendS.o /lib/x86_64-linux-gnu/crtn.o
/usr/bin/ld: cannot find -lstdc++: No such file or directory
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```


In docker:
```
 /usr/lib/llvm-14/bin/clang++ -H test.cc  -v
Ubuntu clang version 14.0.0-1ubuntu1.1
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/lib/llvm-14/bin
Found candidate GCC installation: /usr/lib/gcc/x86_64-linux-gnu/11
Selected GCC installation: /usr/lib/gcc/x86_64-linux-gnu/11
Candidate multilib: .;@m64
Selected multilib: .;@m64
 "/usr/lib/llvm-14/bin/clang" -cc1 -triple x86_64-pc-linux-gnu -emit-obj -mrelax-all --mrelax-relocations -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name test.cc -mrelocation-model pic -pic-level 2 -pic-is-pie -mframe-pointer=all -fmath-errno -ffp-contract=on -fno-rounding-math -mconstructor-aliases -funwind-tables=2 -target-cpu x86-64 -tune-cpu generic -mllvm -treat-scalable-fixed-error-as-warning -debugger-tuning=gdb -v -H -sys-header-deps -fcoverage-compilation-dir=/home/yijiem/grpc -resource-dir /usr/lib/llvm-14/lib/clang/14.0.0 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/x86_64-linux-gnu/c++/11 -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/backward -internal-isystem /usr/lib/llvm-14/lib/clang/14.0.0/include -internal-isystem /usr/local/include -internal-isystem /usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include -internal-externc-isystem /usr/include/x86_64-linux-gnu -internal-externc-isystem /include -internal-externc-isystem /usr/include -fdeprecated-macro -fdebug-compilation-dir=/home/yijiem/grpc -ferror-limit 19 -fgnuc-version=4.2.1 -fcxx-exceptions -fexceptions -fcolor-diagnostics -faddrsig -D__GCC_HAVE_DWARF2_CFI_ASM=1 -o /tmp/test-47b8c2.o -x c++ test.cc
clang -cc1 version 14.0.0 based upon LLVM 14.0.0 default target x86_64-pc-linux-gnu
ignoring nonexistent directory "/usr/lib/gcc/x86_64-linux-gnu/11/../../../../x86_64-linux-gnu/include"
ignoring nonexistent directory "/include"
#include "..." search starts here:
#include <...> search starts here:
 /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11
 /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/x86_64-linux-gnu/c++/11
 /usr/lib/gcc/x86_64-linux-gnu/11/../../../../include/c++/11/backward
 /usr/lib/llvm-14/lib/clang/14.0.0/include
 /usr/local/include
 /usr/include/x86_64-linux-gnu
 /usr/include
End of search list.
 "/usr/bin/ld" -pie -z relro --hash-style=gnu --build-id --eh-frame-hdr -m elf_x86_64 -dynamic-linker /lib64/ld-linux-x86-64.so.2 -o a.out /lib/x86_64-linux-gnu/Scrt1.o /lib/x86_64-linux-gnu/crti.o /usr/lib/gcc/x86_64-linux-gnu/11/crtbeginS.o -L/usr/lib/gcc/x86_64-linux-gnu/11 -L/usr/lib/gcc/x86_64-linux-gnu/11/../../../../lib64 -L/lib/x86_64-linux-gnu -L/lib/../lib64 -L/usr/lib/x86_64-linux-gnu -L/usr/lib/../lib64 -L/usr/lib/llvm-14/bin/../lib -L/lib -L/usr/lib /tmp/test-47b8c2.o -lstdc++ -lm -lgcc_s -lgcc -lc -lgcc_s -lgcc /usr/lib/gcc/x86_64-linux-gnu/11/crtendS.o /lib/x86_64-linux-gnu/crtn.o
/usr/bin/ld: /lib/x86_64-linux-gnu/Scrt1.o: in function `_start':
(.text+0x1b): undefined reference to `main'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```

**Comment by tapetersen** [comment]
Ok for what it's worth it's narrowed down to if `gcc-12` is installed. Adding that to the Dockerimage triggers is (I also changed to 22.04)

```
FROM ubuntu:22.04

ENV DEBIAN_FRONTEND=noninteractive

RUN apt-get -y update && apt-get install -y git wget curl
RUN apt-get install -y build-essential autoconf libtool pkg-config cmake clang libc++-dev gcc-12
RUN wget https://github.com/bazelbuild/bazelisk/releases/download/v1.22.1/bazelisk-linux-amd64
RUN chmod 755 /bazelisk-linux-amd64
RUN mv /bazelisk-linux-amd64 /usr/bin/bazelisk

RUN groupadd -g 1234 yijiem && \
    useradd -m -u 1234 -g yijiem yijiem

USER yijiem
WORKDIR /home/yijiem

RUN git clone https://github.com/grpc/grpc.git

WORKDIR /home/yijiem/grpc
RUN git checkout v1.67.1
RUN git submodule update --init

USER root
RUN apt-get install sudo && gpasswd -a yijiem sudo

CMD ["/bin/bash"]
```

**Comment by yijiem** [comment]
This seems like the same issue as: https://bugs.launchpad.net/ubuntu/+source/llvm-toolchain-14/+bug/2009223. See the last comment in that post:

> This problem happens when you install "gcc-12" package.
If you install "g++-12" package instead, there is no problem.
"gcc-12" package does not install C++ standard library, while "g++-12" package does install it.
